### PR TITLE
A batch of life-cycle examples for different modules

### PR DESCRIPTION
**Pull request by gheber**
Preview at [HDF5 Reference Manual](http://doxygen.hdf5.s3-website.us-east-2.amazonaws.com/_r_m.html)

### CODE REVIEW

File: doxygen/examples/H5PL_examples.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    //! <!-- [create] -->
+    {
+        __label__ fail_set;
+        // enable ONLY filter plugins
+        if (H5PLset_loading_state(H5PL_FILTER_PLUGIN) < 0) {
+            ret_val = EXIT_FAILURE;
+            goto fail_set;
+        }
+
+        // ensure that "/tmp" is a the front of the search path list
```
Comment (ChristopherHogan) [inline doxygen/examples/H5PL_examples.c]: "is a" -> "is at"?

**Review by ChristopherHogan — APPROVED** [review]

File: doxygen/examples/H5PL_examples.c (hunk 1 above)
Comment (gheber) [reply doxygen/examples/H5PL_examples.c]: :+1:

File: configure.ac [hunk 2]
```diff
@@ -1102,9 +1102,11 @@ if test "X$HDF5_DOXYGEN" = "Xyes"; then
   AC_SUBST([DOXYGEN_HTML_FOOTER])
   AC_SUBST([DOXYGEN_HTML_EXTRA_STYLESHEET])
   AC_SUBST([DOXYGEN_HTML_EXTRA_FILES])
+  AC_SUBST([DOXYGEN_TAG_FILE])
```
Comment (byrnHDF) [inline configure.ac:1105]: These same changes need to be made to the CMake locations.

**Review by byrnHDF — CHANGES_REQUESTED** [review]
There probably needs to be a Release.txt entry.
And we need to stop creating these large PRs. There are at least three ways to divide this; the configuration file, the source files and the dox files.

**Comment by byrnHDF** [comment]
Can all these changes be applied to 1.12 and 1.10 branches? Other then obvious version sections?

File: configure.ac (hunk 2 above)
Comment (gheber) [reply configure.ac:1105]: Done.

Comment (byrnHDF) [reply configure.ac:1105]: The CMake files are in c++/src/CMakeLists.txt and src/CMakeLists.txt. I can do that afterwards if you want.

Comment (byrnHDF) [reply configure.ac:1105]: Don't forget DOXYGEN_PREDEFINED

Comment (byrnHDF) [reply configure.ac:1105]: I think the exact same changes can be made to c++/src/CMakeLists.txt.

Comment (byrnHDF) [reply configure.ac:1105]: And DOXYGEN_EXAMPLES_DIRECTORY.

**Review by byrnHDF — APPROVED** [review]
I will rework CMake doxygen after this is merged.

**Review by lrknox — APPROVED** [review]

**Pull request merged**